pandas/tests/arrays/masked/test_arrow_compat.py

```diff
@@ -39,6 +55,30 @@ def test_arrow_roundtrip(data):
     tm.assert_frame_equal(result, df)
 
 
+@td.skip_if_no("pyarrow")
+def test_dataframe_from_arrow_type_mapper(int_dtype):
+    pyarrow = pytest.importorskip("pyarrow", minversion="3.0.0")
+
+    def types_mapper(arrow_type):
+        if pyarrow.types.is_boolean(arrow_type):
+            return pd.BooleanDtype()
+        elif pyarrow.types.is_integer(arrow_type):
+            return int_dtype
+
+    bools_array = pyarrow.array([True, None, False], type=pyarrow.bool_())
+    ints_array = pyarrow.array([1, None, 2], type=pyarrow.int64())
+    record_batch = pyarrow.RecordBatch.from_arrays(
+        [bools_array, ints_array], ["bools", "ints"]
+    )
+    result = record_batch.to_pandas(date_as_object=False, types_mapper=types_mapper)
```

Comment:
Is the `date_as_object` relevant here?

Reply:
No. Removed.

That was leftover from a similar test I was writing for the [db-dtypes package](https://github.com/googleapis/python-db-dtypes-pandas)